Sources/NIO/EventLoopFuture.swift

```diff
@@ -986,6 +970,39 @@ extension EventLoopFuture {
     }
 }
 
+// MARK: andAll
+
+extension EventLoopFuture {
+    /// Returns a new `EventLoopFuture` that succeeds only if all of the provided futures succeed.
+    ///
+    /// This method acts as a successful completion notifier - values fulfilled by each future are discarded.
+    ///
+    /// The returned `EventLoopFuture` fails as soon as any of the provided futures fail.
+    ///
+    /// If it is desired to always succeed, regardless of failures, use `andAllComplete` instead.
+    /// - Parameters:
+    ///     - futures: An array of homogenous `EventLoopFutures`s to wait for.
+    ///     - on: The `EventLoop` on which the new `EventLoopFuture` callbacks will execute on.
+    /// - Returns: A new `EventLoopFuture` that waits for the other futures to succeed.
+    public static func andAllSucceed(_ futures: [EventLoopFuture<Value>], on eventLoop: EventLoop) -> EventLoopFuture<Void> {
+        return .reduce((), futures, eventLoop: eventLoop) { _, _ in }
+    }
+
+    /// Returns a new `EventLoopFuture` that succeeds when all of the provided `EventLoopFuture`s complete.
+    ///
+    /// The returned `EventLoopFuture` always succeeds, acting as a completion notification.
+    /// Values fulfilled by each future are discarded.
+    ///
+    /// If the results are needed, use `whenAllComplete` instead.
+    /// - Parameters:
+    ///     - futures: An array of homogenous `EventLoopFuture`s to wait for.
+    ///     - on: The `EventLoop` on which the new `EventLoopFuture` callbacks will execute on.
+    /// - Returns: A new `EventLoopFuture` that succeeds after all futures complete.
+    public static func andAllComplete(_ futures: [EventLoopFuture<Value>], on eventLoop: EventLoop) -> EventLoopFuture<Void> {
+        return whenAllComplete(futures, eventLoop: eventLoop).map { _ in }
```

Comment:
I think the new implementation should be better suited.

I shifted much of the underlying implementation into `whenAllComplete` from `_whenAllComplete0` that caused your concerns so that now the method is more usable in different contexts.

Let me know what you think.